src/components/Topics.jsx

```diff
@@ -4,16 +4,26 @@ import TopicCard from "./TopicCard";
 
 const Topics = () => {
   const [topics, setTopics] = useState([]);
-
   const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
 
   useEffect(() => {
-    getAllTopics().then((response) => {
-      setTopics(response);
-      setIsLoading(false);
-    });
+    getAllTopics()
+      .then((response) => {
+        setTopics(response);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        console.log(err.message);
+        if (err.message === "Network Error") {
+          setError("Failed to load as you are not online.");
+        }
+      });
```

Comment:
Great that you're getting these topics via an API call rather than hardcoding 🙌 
Nice implementation of loading state